From 471892e1412dd8d5bb4041568ccf819cd6730b2b Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Mon, 28 Sep 2015 17:58:31 +0800
Subject: [PATCH 07/27] dma: dw: remove ->shutdown() hook

Don't touch DMA register as LPSS maybe power gated in ->shutdown() hook.

Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/dma/dw/platform.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/dma/dw/platform.c b/drivers/dma/dw/platform.c
index 8e6eb91..95bff20 100644
--- a/drivers/dma/dw/platform.c
+++ b/drivers/dma/dw/platform.c
@@ -240,15 +240,6 @@ static int dw_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static void dw_shutdown(struct platform_device *pdev)
-{
-	struct dw_dma_chip *chip = platform_get_drvdata(pdev);
-
-	dw_dma_disable(chip);
-	if (chip->clk)
-		clk_disable_unprepare(chip->clk);
-}
-
 #ifdef CONFIG_OF
 static const struct of_device_id dw_dma_of_id_table[] = {
 	{ .compatible = "snps,dma-spear1340" },
@@ -310,7 +301,6 @@ static const struct dev_pm_ops dw_dev_pm_ops = {
 static struct platform_driver dw_driver = {
 	.probe		= dw_probe,
 	.remove		= dw_remove,
-	.shutdown       = dw_shutdown,
 	.driver = {
 		.name	= DRV_NAME,
 		.pm	= &dw_dev_pm_ops,
-- 
2.9.3

